perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -644,6 +645,14 @@ sub setup_categories_and_bodies : Private {
             my $metas = $contact->get_metadata_for_input;
             $category_extras{$contact->category} = $metas if @$metas;
 
+            if ($c->cobrand->can('council_id')) {
```

Comment:
`for_bodies` handles your second point, see https://github.com/mysociety/fixmystreet/blob/515362a13bcaff003143df6d455251fb1c58f913/perllib/FixMyStreet/DB/ResultSet/DefectType.pm#L13. That said, it would make sense to cut down on the number of queries.